docs/madr/decisions/005-policy-matching.md

```diff
@@ -0,0 +1,636 @@
+  - port: 8082
+    tags:
+      kuma.io/service: web-api
+  - port: 8083
+    tags:
+      kuma.io/service: payments
+```
+
+then policy can select a single "web-api" outbound:
+
+```yaml
+to:
+  - targetRef:
+      kind: Service
+      name: web-api
+    outboundConf: ... # conf for "web-api" outbound
+```
+
+policy can select all outbounds ("backend", "web-api" and "payments"):
+
+```yaml
+to:
+  - targetRef:
+      kind: Mesh
+    outboundConf: ... # conf for all outbounds
+```
+
+policy can select an HTTPRoute:
+
+```yaml
+to:
+  - targetRef:
+      kind: HTTPRoute
+      name: payments-get-info-v1-httproute
+    outboundConf: ... # conf for "/v1/getinfo/" route of "payments" outbound 
+```
+
+When several targetRefs select the same outbound, 
+outboundConfs are merged in the order they're presented in the "to" array:
+
+"backend" outbound is selected by 3 targetRefs
+
+```yaml
+to:
+  - targetRef: 
+      kind: Mesh
+    param1: value1
+  - targetRef: 
+      kind: Service
+      name: backend
+    param1: value2
+    param2: value3
+  - targetRef: 
+      kind: Mesh
+    param2: value4
+```
+
+resulting configuration for "backend" outbound is:
+
+```yaml
+param1: value1 # overrides 'value2' from 'targetRef{kind:Service,name:backend}' 
+param2: value3 # overrides 'value4' from 'targetRef{kind:Mesh}'
+```
+
+#### Inbound policy
+
+All inbound policies contains a "from" array.
+Each element of the array has a "targetRef" that selects source of the traffic
+and has a configuration for selected source:
+
+```yaml
+from:
+  - targetRef: Mesh | Service | ProxyGroup
+    sourceConf: ...
+```
+
+Policy can select traffic from "backend" service:
+
+```yaml
+from:
+  - targetRef:
+      kind: Service
+      name: backend
+    sourceConf: ... # conf for traffic coming from "backend" service
+```
+
+Policy can select traffic from proxies with "kuma.io/zone: us-east" tag:
+
+```yaml
+from:
+  - targetRef:
+      kind: ProxyGroup
+      inline:
+        tags:
+          kuma.io/zone: us-east
+    sourceConf: ... # conf for traffic coming from "us-east" zone
+```
+
+Policy can select traffic from all sources:
+
+```yaml
+from:
+  - targetRef:
+      kind: Mesh
+    sourceConf: ... # conf for all incoming traffic
+```
+
+When several targetRefs select the same source,
+sourceConfs are merged in the order they're presented in the "from" array:
+
+Traffic from "backend" of v2 is selected by 3 targetRefs
+
+```yaml
+to:
+  - targetRef: 
+      kind: ProxyGroup
+      inline:
+        tags:
+          kuma.io/service: backend
+          version: v2
+    param1: value1
+  - targetRef:
+      kind: Service
+      name: backend
+    param1: value2
+    param2: value3
+  - targetRef:
+      kind: Mesh
+    param2: value4
+```
+
+resulting configuration for "backend" outbound is:
+
+```yaml
+param1: value1 # overrides 'value2' from 'targetRef{kind:Service,name:backend}' 
+param2: value3 # overrides 'value4' from 'targetRef{kind:Mesh}'
+```
```

Comment:
Do you mean no difference in merging between `to` and `from`?

Reply:
The difference is what is a final consumer of the configuration. It's easy for outbound policies because the final consumer is an outbound (a real object in Envoy proxy). It's less straightforward for inbound policies because the final consumer is a source (an artificial object, different sources represent different clients).